source/val/validation_state.cpp

```diff
@@ -829,4 +845,20 @@ const std::vector<uint32_t>& ValidationState_t::FunctionEntryPoints(
   }
 }
 
+std::string ValidationState_t::Disassemble(const Instruction& inst) const {
+  const spv_parsed_instruction_t& c_inst(inst.c_inst());
+  return Disassemble(c_inst.words, c_inst.num_words);
+}
+
+std::string ValidationState_t::Disassemble(const uint32_t* words,
+                                           uint16_t num_words) const {
+  uint32_t disassembly_options = SPV_BINARY_TO_TEXT_OPTION_NO_HEADER |
+                                 SPV_BINARY_TO_TEXT_OPTION_FRIENDLY_NAMES |
+                                 SPV_BINARY_TO_TEXT_OPTION_SHOW_BYTE_OFFSET;
+
+  return spvtools::spvInstructionBinaryToText(context()->target_env, words,
```

Comment:
Let's fix this in a subsequent CL, but this can be wrong for things like 
`OpConstant %int 42` because you need type info to know that %int is actually an int, and therefore how to interpret the "42" word.  (E.g. compare with a floating point constant.)
A second bad case is OpSwitch where again you need type info to know how many words the literals are.

Interestingly, the spv_parsed_instruction_t has the necessary info.

An actually correct solution would generalize the disassembler so that you pass in the entire binary in and ask only for a specific instruction. It would have to run through the entire binary to collect the type info, and only accumulate string output for the requested instruction (by index).

And yeah, I messed this up on my original attempt. It was quick and dirty and only 95% right.

Reply:
So, is the problem that sometimes it would decompile one of:
```
OpConstant %int 42
OpConstant %int 42.0
```
because it doesn't know if it's a float or an int?

Reply:
Yes, that's one problem.

Another is with 

`OpSwitch %value %default  12 %label0  43 %label1 52 %label3`

The %value and literals are supposed to be integer scalars.  But you need to see the OpTypeInt to know how many bits wide it is, and therefore how many words wide it is.  That's needed to know how to decode the words for the cases.  E.g. 6 words could be 3 pairs of (32-bit literal, target-id) or 2 pairs of (64bit literal, target-id)